geom/alg_convex_hull.go

```
@@ -195,3 +180,36 @@ func distanceSq(p, q XY) Scalar {
 	pSubQ := p.Sub(q)
 	return pSubQ.Dot(pSubQ)
 }
+
+func collinearLine(pts []XY) (Geometry, bool) {
+	if len(pts) < 2 {
+		return nil, false
+	}
+	ps := make([]XY, len(pts))
+	copy(ps, pts)
+
+	startPoint := ps[ltl(ps)]
+	// check collinear
+	sort.Slice(ps, func(i, j int) bool {
+		return distanceSq(startPoint, ps[i]).LT(distanceSq(startPoint, ps[j]))
+	})
+
+	if ps[0].Equals(ps[len(ps)-1]) {
+		return nil, false
+	}
+
+	for i := 1; i < len(ps); i++ {
+		if orientation(ps[0], ps[len(ps)-1], ps[i]) != collinear {
+			return nil, false
+		}
+	}
+
+	// already check that if the initial point is as same as the end point.
+	// Ignore error here.
+	cl, _ := NewLineC(
```

What are your thoughts on panicking instead of ignoring the error?